src/app/components/pages/Concepts.tsx

```diff
@@ -14,41 +14,72 @@ import { isFullyDefinedContext, useUrlPageTheme } from "../../services/pageConte
-    };
+    const [query, filters] = useMemo(() => {
+        const queryParsed = searchParsed.query || null;
+        const query = Array.isArray(queryParsed) ? queryParsed.join(",") : queryParsed;
     
-    const applicableTags = pageContext?.subject ? tags.getDirectDescendents(subjectToTagMap[pageContext.subject]) : tags.allFieldTags;
-    
-    const tagCounts : Record<string, number> = [...applicableTags, ...(pageContext?.subject ? [tags.getById(pageContext?.subject as TAG_ID)] : [])].reduce((acc, t) => ({...acc, [t.id]: listConceptsQuery?.data?.results?.filter(c => c.tags?.includes(t.id)).length || 0}), {});
+        const filterParsed = searchParsed.types || null;
+        const filters = Array.isArray(filterParsed) ? filterParsed.filter(x => !!x) as string[] : filterParsed?.split(",") ?? [];
+        return [query, filters];
+    }, [searchParsed]);
 
-    const searchParsed = queryString.parse(location.search);
+    const applicableTags = pageContext?.subject 
+        // this includes all subject tags and all field tags
+        ? [tags.getById(subjectToTagMap[pageContext.subject]), ...tags.getDirectDescendents(subjectToTagMap[pageContext.subject])]
+        : [...tags.allSubjectTags, ...tags.allFieldTags];
 
-    const queryParsed = searchParsed.query || "";
-    const query = queryParsed instanceof Array ? queryParsed[0] : queryParsed;
+    const [searchText, setSearchText] = useState(query);
+    const [conceptFilters, setConceptFilters] = useState<Tag[]>(
+        applicableTags.filter(f => filters.includes(f.id))
+    );
+    const [shortcutResponse, setShortcutResponse] = useState<ShortcutResponse[]>();
 
-    const filterParsed = (searchParsed.types || (TAG_ID.physics + "," + TAG_ID.maths + "," + TAG_ID.chemistry + "," + TAG_ID.biology));
-    const filters = (Array.isArray(filterParsed) ? filterParsed[0] || "" : filterParsed || "").split(",");
+    const listConceptsQuery = useListConceptsQuery(pageContext 
+        ? {conceptIds: undefined, tagIds: pageContext?.subject ?? tags.allSubjectTags.map(t => t.id).join(",")}
+        : skipToken
+    );
 
-    const [searchText, setSearchText] = useState(query);
-    const [conceptFilters, setConceptFilters] = useState<Tag[]>([
-        ...(pageContext?.subject ? [tags.getById(subjectToTagMap[pageContext.subject])] : []),
-        ...applicableTags.filter(f => filters.includes(f.id))
-    ]);
+    const shortcutAndFilter = (concepts?: ContentSummaryDTO[], excludeTopicFiltering?: boolean) => {
+        const searchResults = concepts?.filter(c =>
+            matchesAllWordsInAnyOrder(c.title, searchText || "") ||
+            matchesAllWordsInAnyOrder(c.summary, searchText || "")
+        );
+        
+        const filteredSearchResults = searchResults
+            ?.filter((result) => excludeTopicFiltering || !filters.length || result?.tags?.some(t => filters.includes(t)))
+            .filter((result) => !pageContext?.stage || isRelevantToPageContext(result.audience, pageContext))
```

Comment:
```suggestion
            .filter((result) => !pageContext?.stage || !(pageContext?.stage.length > 0) || isRelevantToPageContext(result.audience, pageContext))
```

The generic concepts page isn't showing any concepts since stage is defined but empty.

Reply:
Forgot to run with disabled cache 😣 nice spot. We can simplify to just `!pageContext?.stage?.length` though.